app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdEffectHandler.kt

```diff
@@ -38,8 +38,25 @@ class ScanSimpleIdEffectHandler @AssistedInject constructor(
       .addConsumer(OpenPatientSummary::class.java, ::openPatientSummary, schedulersProvider.ui())
       .addConsumer(OpenPatientSearch::class.java, ::openPatientSearch, schedulersProvider.ui())
       .addTransformer(OnlinePatientLookupWithIdentifier::class.java, onlinePatientLookupWithIdentifier())
+      .addTransformer(SaveCompleteMedicalRecords::class.java, saveCompleteMedicalRecords())
       .build()
 
+  private fun saveCompleteMedicalRecords(): ObservableTransformer<SaveCompleteMedicalRecords, ScanSimpleIdEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(schedulersProvider.io())
+          .map(SaveCompleteMedicalRecords::completeMedicalRecords)
+          .map { completeMedicalRecords ->
+            completeMedicalRecords to completeMedicalRecords.map { medicalRecord ->
+              patientRepository.saveCompleteMedicalRecord(medicalRecord)
+            }
```

Comment:
Maybe in another PR, we can change the signature of `PatientRepository#saveCompleteMedicalRecord` to accept a `List<CompleteMedicalRecord>` so we don't need this additional iteration. Would be more efficient as well since we can merge all the records in a single transaction.